src/main/java/io/codelex/oop/cars/CarService.java

```diff
@@ -0,0 +1,177 @@
+            }
+        }
+    }
+
+    public void sortedByPrice(String condition) {
+        if (condition.equals("ascending")) {
+            List<Car> ascendingPrice = cars.stream()
+                    .sorted(Comparator.comparing(Car::getPrice)).toList();
+            System.out.println("Cars sorted by price ascending:");
+            for (Car car : ascendingPrice) {
+                System.out.println(car.getName() + " " + car.getModel() + " " + car.getPrice());
+            }
+        } else if (condition.equals("descending")) {
+            List<Car> descendingPrice = cars.stream()
+                    .sorted(Comparator.comparing(Car::getPrice).reversed()).
+                    toList();
+            System.out.println("Cars sorted by price descending:");
+            for (Car car : descendingPrice) {
+                System.out.println(car.getName() + " " + car.getModel() + " " + car.getPrice() + " eur");
+            }
+        } else {
+            System.out.println("Invalid input argument!");
+        }
+    }
+
+    public void sortedByYearOfManufacture(String condition) {
+        if (condition.equals("ascending")) {
+            List<Car> ascendingYear = cars.stream()
+                    .sorted(Comparator.comparing(Car::getYearOfManufacture)).toList();
+            System.out.println("Cars sorted by year of manufacture ascending:");
+            for (Car car : ascendingYear) {
+                System.out.println(car.getName() + " " + car.getModel() + " " + car.getYearOfManufacture());
+            }
+        } else if (condition.equals("descending")) {
+            List<Car> descendingYear = cars.stream()
+                    .sorted(Comparator.comparing(Car::getYearOfManufacture).reversed()).
+                    toList();
+            System.out.println("Cars sorted by year of manufacture descending:");
+            for (Car car : descendingYear) {
+                System.out.println(car.getName() + " " + car.getModel() + " " + car.getYearOfManufacture());
+            }
+        } else {
+            System.out.println("Invalid input argument!");
+        }
+    }
+
+    public boolean checkIfCarIsOnList(Car c) {
+        System.out.print("Current " + c.getName() + " " + c.getModel() + " is in our list: ");
+        return cars.contains(c);
+    }
+
+    public void findByManufacturer(Manufacturer x) {
+        System.out.println("Cars manufactured by " + x.getName() + " manufacture are:");
+        for (Car car : cars) {
+            if (car.getManufacturers().contains(x)) {
+                System.out.println(car.getName() + " " + car.getModel());
+            }
+        }
+    }
+
+    public void carsByManufacturer(Manufacturer x, int year) {
+        System.out.println("Cars manufactured by " + x.getName() + " after " + year + " year are:");
+        int countAfter = 0;
+        for (Car car : cars) {
```

Comment:
šos 3 for loopus noteikti var apvienot vienā

Reply:
Apvienoju vienā. Tagad gan nedaudz savādāk izvada info, bet kods ir īsāks un cerams saprotamāks.